app/src/main/java/com/example/eric/socials/FeedActivity.java

```diff
@@ -0,0 +1,118 @@
+    private ArrayList<Social> mSocials = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_feed);
+
+        recyclerView = (RecyclerView)findViewById(R.id.recylcerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+
+        mDrawerList = (ListView) findViewById(R.id.navList);
+        mAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, optionsArray);
+        mDrawerList.setAdapter(mAdapter);
+
+        mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {                  // Listener for clicking an Item on the Side Drawer
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                SparseBooleanArray clickedItemPositions = mDrawerList.getCheckedItemPositions();
+                mAdapter.notifyDataSetChanged();
+
+                for(int j = 0;j < clickedItemPositions.size(); j ++){
+                    boolean checked = clickedItemPositions.valueAt(j);
+
+                    if(checked){
+                        int key = clickedItemPositions.keyAt(j);
+                        String item = (String) mDrawerList.getItemAtPosition(key);
+
+                        if(item.equalsIgnoreCase(getString(R.string.update_profile_button))){
+                            Intent intent = new Intent(getApplicationContext(), UpdateProfile.class);
+                            startActivity(intent);
+                        }
+                        if(item.equalsIgnoreCase(getString(R.string.logout_button))){
+                            FirebaseAuth.getInstance().signOut();
+                            Intent intent = new Intent(getApplicationContext(), LoginActivity.class);
+                            startActivity(intent);
+                        }
+                        if(item.equalsIgnoreCase(getString(R.string.create_new_social_button))){
+                            Intent intent = new Intent(getApplicationContext(), AddEvent.class);
+                            startActivity(intent);
+                        }
+                    }
+                }
+            }
+        });
+
+        DatabaseReference myRef = FirebaseDatabase.getInstance().getReference(getString(R.string.socials_db_tree));         // DB Reference to the socials tree
+        myRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                mSocials.clear();
+                for (DataSnapshot socialSnapShot: dataSnapshot.getChildren()) {
+                    GenericTypeIndicator<ArrayList<String>> t = new GenericTypeIndicator<ArrayList<String>>() {};
+                    Social social = new Social(socialSnapShot.child(getString(R.string.eventName)).getValue(String.class),
```

Comment:
You can give Firebase enter classes, like snapshot.getValue(Social.class) and it'll populate the fields based on the variable names. It is useful in cases like this.